Lib/test/test_asyncio/test_events.py

```diff
@@ -2658,7 +2659,8 @@ def tearDown(self):
             asyncio.get_event_loop = self.get_event_loop_saved
 
     if sys.platform != 'win32':
-
+        @unittest.skipUnless(os.path.exists("/dev/shm") and stat.S_IMODE(os.stat('/dev/shm').st_mode) >= 0o777,
+                             "Doesn't work without /dev/shm.")
```

Comment:
I'm not sure if /dev/shm should be hardcoded here. This check looks to be specific to Linux, no?

Maybe use Arena._dir_candidates of multiprocessing.heap. I'm not sure what to do if _dir_candidates is empty (platforms other than Linux).

I dislike that test_asyncio becomes tidely coupled to multiprocessing internals. Is there a function in multiprocessing somewhere to check if /dev/shm is required and usable?

Reply:
Instead of that, why not skip if there is an environment variabled named e.g. `PYTHON_NO_DEV_SHM`?

Reply:
@pitrou I would still hoped for some kind of magic, what @vstinner suggested. Any ideas?

Reply:
I think it's more flexible to let you handle the magic in your build configuration. I don't think we want to handle all weird corner cases in the test suite.

Reply:
OK, so the patch with ``unittest.skipIf('PYTHON_NO_DEV_SHM' in os.environ)`` would be accepted?

Reply:
IMHO it should. I'll let @vstinner comment.
(or perhaps `os.environ.get('PYTHON_NO_DEV_SHM') == '1'`)

Reply:
I don't understand the relationship between asyncio, multiprocessing and /dev/shm.